numpy/_core/src/multiarray/stringdtype/static_string.c

```diff
@@ -774,7 +743,9 @@ NpyString_pack(npy_string_allocator *allocator,
 NPY_NO_EXPORT int
 NpyString_share_memory(const npy_packed_static_string *s1, npy_string_allocator *a1,
                        const npy_packed_static_string *s2, npy_string_allocator *a2) {
-    if (a1 != a2 || is_not_a_vstring(s1) || is_not_a_vstring(s2)) {
+    if (a1 != a2 ||
+        is_short_string(s1) || is_short_string(s2) ||
```

Comment:
I could see the case for reintroducing `is_a_vstring` here just to negate it. But with use in just a single function, it seemed fine to just write it out.